sqlmesh/core/audit/definition.py

```diff
@@ -247,22 +261,173 @@ def macro_definitions(self) -> t.List[d.MacroDef]:
         """All macro definitions from the list of expressions."""
         return [s for s in self.expressions if isinstance(s, d.MacroDef)]
 
-    def _create_query_renderer(self, model: _Model) -> QueryRenderer:
+    def _create_query_renderer(self, node: _Node) -> QueryRenderer:
+        kwargs: t.Dict[str, t.Any] = {}
+
+        node_dialect = ""
+        if node.is_model:
+            node = t.cast(_Model, node)
+            node_dialect = node.dialect
+            kwargs = {
+                "only_execution_time": node.kind.only_execution_time,
+                "python_env": node.python_env,
+            }
+        elif node.is_audit:
+            node = t.cast(StandaloneAudit, node)
+            node_dialect = node.dialect
+            kwargs = {"python_env": node.python_env}
+
         return QueryRenderer(
             self.query,
-            self.dialect or model.dialect,
+            self.dialect or node_dialect,
             self.macro_definitions,
             path=self._path or Path(),
             jinja_macro_registry=self.jinja_macros,
-            python_env=model.python_env,
-            only_execution_time=model.kind.only_execution_time,
+            **kwargs,
         )
 
 
+class StandaloneAudit(_Node):
+    """
+    Args:
+        name: The unique name of the standalone audit.
+        owner: The owner of the standalone audit.
+        description: The optional standalone audit description.
+        stamp: An optional arbitrary string sequence used to create new standalone audit versions without making
+            changes to any of the functional components of the definition.
+        tags:
+        hash_raw_query:
+    """
+
+    name: str
+    depends_on_: t.Optional[t.Set[str]] = Field(default=None, alias="depends_on")
+    audit: Audit
+    tags: t.List[str] = []
+    hash_raw_query: bool = False
+
+    python_env_: t.Optional[t.Dict[str, Executable]] = Field(default=None, alias="python_env")
+    source_type: Literal["audit"] = "audit"
+
+    _path: Path = Path()
+    _depends_on: t.Optional[t.Set[str]] = None
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_fields(
+        cls, values: t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]
```

Comment:
I think this type signature is useless to the point that you're better off just using `t.Dict[str, t.Any]`. Nobody can decipher this anyway. Same for return value.